File: frontend/store/interface/member.ts
export class Member {
  id: number;
  createdDate: Date;
  isDeleted: boolean;
  email: string;
  nickname: string;
  isAdmin: boolean;
  isKakao: boolean;
  birthday: string;
  gender: string;
  mbti: string;
  interest: string;
  likelist: string;
  banlist: string;
  image: string;
  friendStatus: string;

  constructor(data: any) {
    this.id = data.id || 0;
    this.createdDate = data.createdDate || Date.now();
    this.isDeleted = data.isDeleted || false;
    this.email = data.email || "";
    this.nickname = data.nickname || "";
    this.isAdmin = data.isAdmin || false;
    this.isKakao = data.isKakao || false;
    this.birthday = data.birthday || "";
    this.gender = data.gender || "";
    this.mbti = data.mbti || "";
    this.interest = data.interest || "";
    this.likelist = data.likelist || "";
    this.banlist = data.banlist || "";
    this.image = data.image || "";
    this.friendStatus = data.friendStatus || "";
  }
}

function MemberList(data: Array<any>): Member[] {
  return data.map((people) => {
    return new Member(people);
  });
}

export default MemberList;

File: backend/src/main/java/com/seonbi/db/entity/Member.java
package com.seonbi.db.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
@Getter
@Setter
@ToString
public class Member extends BaseEntity{
    // Wrapper 타입인 Long이나 Integer를 쓰면 id가 없는 경우엔 확실히 null이고, 그 자체로 id가 없다는걸 보장함.

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY) // auto increment
    private Long memberId;
    private String email;
    private String nickname;
    private Boolean isAdmin=false;
    private Boolean isKakao=false;
    private String birthday;    // 1997.04.10
    @JsonIgnore
    private String password;
    private String gender;
    private String mbti;
    private String interest;
    private String likelist;
    private String banlist;
    private Long imageId=0l;    // default는 기본 이미지
    private String verse;

}
